omnigibson/object_states/on_top.py

```diff
@@ -33,4 +35,5 @@ def _get_value(self, other):
             return False
 
         adjacency = self.obj.states[VerticalAdjacency].get_value()
-        return other in adjacency.negative_neighbors and other not in adjacency.positive_neighbors
+        other_adjacency = other.states[VerticalAdjacency].get_value()
+        return other in adjacency.negative_neighbors and other not in adjacency.positive_neighbors and self.obj not in other_adjacency.negative_neighbors
```

Comment:
nit: maybe add a comment to explain why this is needed?

Reply:
What about Inside and Under?

Reply:
fixed for Under